okhttp/src/main/java/io/grpc/okhttp/Headers.java

```diff
@@ -45,8 +46,8 @@
    * creating a stream. Since this serializes the headers, this method should be called in the
    * application thread context.
    */
-  public static List<Header> createRequestHeaders(Metadata headers, String defaultPath,
-      String authority, String userAgent) {
+  public static List<Header> createRequestHeaders(
```

Comment:
Whoa. This class needs to be made non-public or marked as `@Internal`. I'll send out a PR that makes it package-private. That's safe since Header is internal, so there's no way to use this API safely already. Eric, no need to maintain compatibility or add the annotations for now.